cms/djangoapps/maintenance/views.py

```diff
@@ -211,3 +226,80 @@ def post(self, request):
             exc_info=True
         )
         return self.render_response()
+
+
+class AnnouncementIndexView(ListView, MaintenanceBaseView):
+    """
+    View for viewing the announcements shown on the dashboard, used by the global staff.
+    """
+
+    model = Announcement
+    object_list = Announcement.objects.all()
+    context_object_name = 'announcement_list'
+    paginate_by = 9
+
+    def __init__(self):
+        super(AnnouncementIndexView, self).__init__(MAINTENANCE_VIEWS['announcement_index'])
+
+    @method_decorator(require_global_staff)
+    def get(self, request, *args, **kwargs):
+        if "limit" in request.GET.keys():
+            self.paginate_by = request.GET.get("limit")
```

Comment:
Need to enforce an upper limit here, else people could bring the database to its knees by creating and requesting millions of announcements on a page.

Reply:
I added `max(..., 50)`. Does that seem like a reasonable limit? Perhaps it should be a bit higher even..